src/pages/workspace/WorkspaceNewRoomPage.js

```diff
@@ -77,50 +78,107 @@ class WorkspaceNewRoomPage extends React.Component {
     }
 
     /**
-     * Called when a workspace is selected. Also calls checkAndModifyRoomName,
-     * which displays an error if the given roomName exists on the newly selected workspace.
-     * @param {String} policyID
+     * Called when the "Create Room" button is pressed.
      */
-    onWorkspaceSelect(policyID) {
-        this.setState({policyID});
-        this.checkAndModifyRoomName(this.state.roomName);
+    onSubmit() {
+        if (!this.validate()) {
+            return;
+        }
+        Report.createPolicyRoom(
+            this.state.policyID,
+            this.state.roomName,
+            this.state.visibility,
+        );
     }
 
     /**
-     * Called when the "Create Room" button is pressed.
+     * Called when the onSubmit() method is triggered.
+     * which validates the form fields.
+     * @returns {Boolean}
      */
-    onSubmit() {
-        Report.createPolicyRoom(this.state.policyID, this.state.roomName, this.state.visibility);
+    validate() {
+        const roomName = this.state.roomName.trim();
+        const policyID = this.state.policyID;
+        const visibility = this.state.visibility;
+        const errors = {};
+
+        if (!roomName || roomName === '#') {
+            errors.roomName = 'Please enter a room name';
+        } else if (
+            policyID
+            && this.checkAndModifyRoomName(roomName).isExistingRoomName
+        ) {
+            errors.roomName = this.props.translate('newRoomPage.roomAlreadyExists');
+        } else {
+            errors.roomName = '';
+        }
+
+        if (!policyID) {
+            errors.policyID = 'Please select a workspace';
+        } else {
+            errors.policyID = '';
+        }
+
+        if (!visibility) {
+            errors.visibility = 'Please select a visibility';
+        } else {
+            errors.visibility = '';
+        }
+
+        if (!errors.roomName && !errors.policyID && !errors.visibility) {
+            this.setState({errors: {}});
+            return true;
+        }
+
+        this.setState(prevState => ({
+            errors: {...prevState.errors, ...errors},
+        }));
+        return false;
+    }
+
+    /**
+     * Clear the errors associated to inputKey if found and store the inputKey new value in the state.
+     *
+     * @param {String} inputKey
+     * @param {String} value
+     */
+    clearErrorAndSetValue(inputKey, value) {
+        this.setState(prevState => ({
+            [inputKey]: value,
+            errors: {
+                ...prevState.errors,
+                [inputKey]: false,
```

Comment:
```suggestion
                [inputKey]: '',
```

Keep the same datatype.